features/harmful-apis/main.js

```diff
@@ -0,0 +1,329 @@
+        id: 'push-api',
+        run: () => {
+            return ('ServiceWorkerRegistration' in window) && ('pushManager' in ServiceWorkerRegistration.prototype);
+        }
+    },
+    {
+        id: 'first-party-sets-api',
+        run: () => {
+            if (!window.cookieStore) {
+                return '❌ Can\'t be tested - CookieStore not available';
+            }
+
+            return window.cookieStore.set({ name: 'first-party-sets-test', value: 'value' })
+                .then(() => window.cookieStore.getAll())
+                .then(cookies => {
+                    const cookie = cookies.find(c => c.name === 'first-party-sets-test');
+
+                    return ('sameParty' in cookie);
+                });
+        }
+    }
+];
+
+// object that contains results of all tests
+const results = {
+    page: `${PAGE_NAME}-test`,
+    date: null,
+    results: []
+};
+
+function resultToHTML (data) {
+    return ((data === false) ? '❌ API not available' : '✅ API available') + ' (' + JSON.stringify(data, null, 2) + ')';
+}
+
+/**
+ * Test runner
+ */
+function runTests () {
+    startButton.setAttribute('disabled', 'disabled');
+    downloadButton.removeAttribute('disabled');
+    testsDiv.removeAttribute('hidden');
+
+    results.results.length = 0;
+    results.date = (new Date()).toUTCString();
+    let all = 0;
+    let failed = 0;
+
+    testsDetailsDiv.innerHTML = '';
+
+    function updateSummary () {
+        testsSummaryDiv.innerText = `Performed ${all} tests${failed > 0 ? ` (${failed} failed)` : ''}. Click for details.`;
+    }
+
+    for (const test of tests) {
+        const resultObj = {
+            id: test.id,
+            value: null
+        };
+        results.results.push(resultObj);
+
+        const li = document.createElement('li');
+        li.id = `test-${test.id.replace(' ', '-')}`;
+        li.innerHTML = `${test.id} - <span class='value'>…</span>`;
+        const valueSpan = li.querySelector('.value');
+
+        testsDetailsDiv.appendChild(li);
+
+        try {
+            const result = test.run();
+
+            if (result instanceof Promise) {
+                result
+                    .then(data => {
+                        valueSpan.innerHTML = resultToHTML(data);
+                        resultObj.value = data || null;
+                    })
+                    .catch(e => {
+                        failed++;
+                        valueSpan.innerHTML = `❌ error thrown ("${e.message ? e.message : e}")`;
+                        updateSummary();
+                    });
+            } else {
+                valueSpan.innerHTML = resultToHTML(result);
+                resultObj.value = result || null;
+            }
+        } catch (e) {
+            failed++;
+            valueSpan.innerHTML = `❌ error thrown ("${e.message ? e.message : e}")`;
+        }
+
+        all++;
+    }
+
+    updateSummary();
+
+    startButton.removeAttribute('disabled');
+}
+
+function downloadTheResults () {
```

Comment:
The downloaded results show `null` for false. Is that expected?

Reply:
Only for failed tests, for other tests true/false. I fixed it.